tests/stub/driver_execute_query/test_driver_execute_query.py

```diff
@@ -0,0 +1,262 @@
+from nutkit.frontend import (
+    BookmarkManager,
+    Driver,
+    Neo4jBookmarkManagerConfig,
+)
+import nutkit.protocol as types
+from tests.shared import TestkitTestCase
+from tests.stub.shared import StubServer
+
+
+class TestDriverExecuteQuery(TestkitTestCase):
+    required_features = (
+        types.Feature.BOLT_5_0,
+        types.Feature.API_BOOKMARK_MANAGER,
+        types.Feature.API_DRIVER_EXECUTE_QUERY
+    )
+
+    def setUp(self):
+        super().setUp()
+        self._router = StubServer(9000)
+        self._reader = StubServer(9010)
+        self._writer = StubServer(9020)
+        self._driver = None
+
+    def tearDown(self):
+        if self._driver:
+            self._driver.close()
+
+        self._router.reset()
+        self._reader.reset()
+        self._writer.reset()
+        return super().tearDown()
+
+    def test_execute_query_without_params_and_config(self):
+        self._start_server(self._router, "router.script")
+        self._start_server(self._writer, "tx_return_1.script")
+        self._driver = self._new_driver()
+
+        eager_result = self._driver.execute_query("RETURN 1 AS n")
+
+        self.assertEqual(eager_result.keys, ["n"])
+        self.assertEqual(eager_result.records, [
+                         types.Record(values=[types.CypherInt(1)])])
+        self.assertIsNotNone(eager_result.summary)
+
+    def test_execute_query_without_config(self):
+        self._start_server(self._router, "router.script")
+        self._start_server(self._writer, "tx_return_1_with_params.script")
+        self._driver = self._new_driver()
+
+        eager_result = self._driver.execute_query("RETURN 1 AS n", {
+            "a": types.CypherInt(1)
+        })
+
+        self.assertEqual(eager_result.keys, ["n"])
+        self.assertEqual(eager_result.records, [
+                         types.Record(values=[types.CypherInt(1)])])
+        self.assertIsNotNone(eager_result.summary)
+
+    def test_configure_routing_to_writers(self):
+        self._start_server(self._router, "router.script")
+        self._start_server(self._writer, "tx_return_1_with_params.script")
+        self._driver = self._new_driver()
+
+        eager_result = self._driver.execute_query("RETURN 1 AS n", {
+            "a": types.CypherInt(1)
+        }, {"routing": "w"})
+
+        self.assertEqual(eager_result.keys, ["n"])
```

Comment:
Can we fix all the assertions being backwards, I know it's a common issue in testkit but it should be (expected, actual)

Reply:
That's documented nowhere. The order is arbitrary. It's just PyCharm that felt entitled to assign the first argument as the expected value. It's honestly audacious.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertEqual

Reply:
unittest truly is the worst why don't they have this defined. It is my understanding that in most assertion libraries you assert(expect, actual)

Reply:
It was my js doing `expect(this).toEqual(that)`

Reply:
Just ignore PyCharm it's stupid. Here goes the output with pytest
```python
def test_foo(self):
    assert 42 != 42
```
leads to
```
42 != 42

Expected :42
Actual   :42
```
:face_with_head_bandage:

Reply:
I double check the result of a failure in the terminal and it is getting like this:

```
Traceback (most recent call last):
  File "./tests/stub/driver_execute_query/test_driver_execute_query.py", line 261, in test_configure_custom_bookmark_manager
    self.assertEqual(summary2.query_type, "w")
AssertionError: 'r' != 'w'
- r
+ w
```

I will invert everything here